src/test/java/org/assertj/core/internal/throwables/Throwables_assertHasRootCause_Test.java

```diff
@@ -0,0 +1,197 @@
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.mockito.Mockito.verify;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.ThrowablesBaseTest;
+import org.junit.jupiter.api.Test;
+
+public class Throwables_assertHasRootCause_Test extends ThrowablesBaseTest {
+
+  @Test
+  public void should_pass_if_root_cause_has_expected_type_and_message() {
+    // GIVEN
+    Throwable rootCause = new IllegalArgumentException("bang");
+    Throwable expected = new IllegalArgumentException("bang");
+    Throwable throwable = withRootCause(rootCause);
+
+    // WHEN
+    throwables.assertHasRootCause(someInfo(), throwable, expected);
+
+    // THEN
+    // no exception thrown
+  }
+
+  @Test
+  public void should_pass_if_root_cause_is_expected_cause() {
+    // GIVEN
+    Throwable cause = new IllegalArgumentException("wibble");
+    Throwable expected = new IllegalArgumentException("wibble");
+    Throwable throwable = withCause(cause);
+
+    // WHEN
+    throwables.assertHasRootCause(someInfo(), throwable, expected);
+
+    // THEN
+    // no exception thrown
+  }
+
+  @Test
+  public void should_pass_if_actual_has_no_cause_and_expected_cause_is_null() {
+    // GIVEN
+    Throwable throwable = new Throwable();
+    Throwable expected = null;
+
+    // WHEN
+    throwables.assertHasRootCause(someInfo(), throwable, expected);
+
+    // THEN
+    // no exception thrown
+  }
+
+  @Test
+  public void should_fail_if_actual_is_null() {
+    // GIVEN
+    final Throwable throwable = null;
+    final Throwable expected = new Throwable();
+
+    // WHEN
+    AssertionError actual = expectAssertionError(
+      () -> throwables.assertHasRootCause(someInfo(), throwable, expected)
+    );
+
+    // THEN
+    assertThat(actual)
+      .hasMessage(actualIsNull());
+  }
+
+  @Test
+  public void should_fail_if_actual_has_no_cause_and_expected_cause_is_not_null() {
+    // GIVEN
+    final Throwable throwable = new Throwable();
+    final Throwable expected = new RuntimeException();
+    final AssertionInfo info = someInfo();
+
+    // WHEN
+    expectAssertionError(
+      () -> throwables.assertHasRootCause(info, throwable, expected)
+    );
+
+    // THEN
+    verify(failures).failure(info, shouldHaveRootCause(null, expected));
+  }
+
+  @Test
+  public void should_fail_if_actual_has_cause_and_expected_cause_is_null() {
+    // GIVEN
+    Throwable rootCause = new IllegalArgumentException();
+    final Throwable throwable = withRootCause(rootCause);
+    final Throwable expected = null;
+    final AssertionInfo info = someInfo();
+
+    // WHEN
+    expectAssertionError(
+      () -> throwables.assertHasRootCause(info, throwable, expected)
+    );
+
+    // THEN
+    verify(failures).failure(info, shouldHaveNoCause(throwable));
+  }
+
+  @Test
+  public void should_fail_if_cause_is_not_instance_of_expected_type() {
```

Comment:
I think most of the `should_fail...` tests can be combined as a parameterized test for the tests where WHEN and THEN sections are the same.

JUnit 5 comes with great support for parameterized tests, I'm starting to use that more and more.
I suggest following this test example: https://github.com/joel-costigliola/assertj-core/blob/recursive-api/src/test/java/org/assertj/core/internal/objects/Objects_assertIsEqualToUsingRecursiveComparison_Test.java#L219.

Another (minor) improvement would be to extract `final AssertionInfo info = someInfo();` to a constant and use everywhere (in these tests we don't really care about that value as long as it is provided).